src/scenarios/shared/devicepowerconsumption.py

```diff
@@ -0,0 +1,288 @@
+            RunCommand(reconnectYepKitPowerCmd, verbose=True).run() # Make sure the board is connected
+            
+            for i in range(testiterations):
+                # Clear logs
+                RunCommand(clearLogsCmd, verbose=True).run()
+                RunCommand(clearBatteryStatsCmd, verbose=True).run()
+                startStats = RunCommand(androidHelper.startappcommand, verbose=True)
+                startStats.run()
+                preExecutionBatteryInfo = RunCommand(getGeneralBatteryInformation, verbose=True)
+                preExecutionBatteryInfo.run()
+                RunCommand(disconnectYepKitPowerCmd, verbose=True).run()
+                time.sleep(runtimeseconds)
+                RunCommand(reconnectYepKitPowerCmd, verbose=True).run()
+                time.sleep(5) # Wait for the phone to reconnect to the commputer
+                postExecutionBatteryInfo = RunCommand(getGeneralBatteryInformation, verbose=True)
+                postExecutionBatteryInfo.run()
+                RunCommand(androidHelper.stopappcommand, verbose=True).run()
+                
+                captureUid = RunCommand(getUidOfPackageCmd, verbose=True)
+                captureUid.run()
+
+                # Get the Uid from the captureUid output
+                # The output format and target are as follows:
+                # package:com.companyname.mauiandroiddefault uid:1<capture start>1453<end capture>
+                uidSearchString = r"package:" + packagename + r" uid:([0-9]*)"
+                uidCapture = re.search(uidSearchString, captureUid.stdout)
+                if not uidCapture:
+                    raise Exception("Failed to capture the uid!")
+                uid = uidCapture.group(1)
+
+                # Include the uid filter to keep the output smaller
+                captureBatteryStatsCmd = [
+                    androidHelper.adbpath,
+                    'shell',
+                    'dumpsys',
+                    'batterystats',
+                    '--charged',
+                    packagename,
+                    '-c',
+                    '|',
+                    'grep',
+                    uid               
+                ]
+                
+                captureProcStats = RunCommand(captureBatteryStatsCmd, verbose=True)
+                captureProcStats.run()
+
+                capturedValues = {}
+                # Get the mAh estimated power use based on the Uid from the battery stats
+                # Explanation of the 4 groups: https://stackoverflow.com/questions/75390939/android-how-to-interpret-pwi-power-use-item-from-battery-stats-dumpsys
+                # Example output section and target:
+                # 9,11458,l,pwi,uid,0.0820,0,0.134,0.0342 captures total usage, is battery consumer (consult link below), screen usage, and proportional usage
+                totalPowerSearchString = r"[\d]*," + uid + r",l,pwi,uid,([\d]*.?[\d]*),([\d]*.?[\d]*),([\d]*.?[\d]*),([\d]*.?[\d]*)"
```

Comment:
Is there a reason you use `[\d*]` instead of `\d*`. 

Also, I guess, the number matched is "required", thus maybe `\d+` would be a better.